xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -533,12 +533,18 @@ void CActiveAE::StateMachine(int signal, Protocol *port, Message *msg)
         switch (signal)
         {
         case CActiveAEControlProtocol::RECONFIGURE:
+        {
           if (m_streams.empty())
           {
             streaming = false;
             m_sink.m_controlPort.SendOutMessage(CSinkControlProtocol::STREAMING, &streaming, sizeof(bool));
           }
+          AudioSettings settings = m_settings;
           LoadSettings();
+          if (settings.streamNoise != m_settings.streamNoise)
```

Comment:
what the the magic here? first you assign m_settings to settings, than you compare those two. what do you expect? what behaviour do you want to achieve? if you followed my advice with naming of the signals, you would have avoided this case. when "setting" a value the old value is not of any interest. you can always send the message and no comparison is needed.